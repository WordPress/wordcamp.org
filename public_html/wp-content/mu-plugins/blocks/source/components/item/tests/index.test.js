/* global describe, test, expect */
/**
 * External dependencies
 */
import renderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import { ItemTitle } from '../';

describe( 'ItemTitle', () => {
	test( 'should render a heading tag with the default level.', () => {
		const component = renderer.create( <ItemTitle title="Example Title" /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render a heading tag of level 1.', () => {
		const component = renderer.create( <ItemTitle title="Example Title" headingLevel={ 1 } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render a heading tag with a custom class.', () => {
		const component = renderer.create( <ItemTitle title="Example Title" className="my-test-heading" /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render a heading tag with a set alignment.', () => {
		const component = renderer.create( <ItemTitle title="Example Title" align="right" /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render a heading tag with a link.', () => {
		const component = renderer.create( <ItemTitle title="Example Title" link="https://wordpress.com" /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render a heading tag with a link, custom class name, heading level, and alignment.', () => {
		const component = renderer.create(
			<ItemTitle
				title="Example Title"
				headingLevel={ 1 }
				className="my-test-heading"
				align="right"
				link="https://wordpress.com"
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
