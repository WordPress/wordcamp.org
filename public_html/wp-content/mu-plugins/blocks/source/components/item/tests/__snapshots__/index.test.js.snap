// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemTitle should render a heading tag of level 1. 1`] = `
<h1
  className="wordcamp-block__item-title"
  style={Object {}}
>
  Example Title
</h1>
`;

exports[`ItemTitle should render a heading tag with a custom class. 1`] = `
<h3
  className="wordcamp-block__item-title my-test-heading"
  style={Object {}}
>
  Example Title
</h3>
`;

exports[`ItemTitle should render a heading tag with a link, custom class name, heading level, and alignment. 1`] = `
<h1
  className="wordcamp-block__item-title my-test-heading"
  style={
    Object {
      "textAlign": "right",
    }
  }
>
  <a
    href="https://wordpress.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    Example Title
  </a>
</h1>
`;

exports[`ItemTitle should render a heading tag with a link. 1`] = `
<h3
  className="wordcamp-block__item-title"
  style={Object {}}
>
  <a
    href="https://wordpress.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    Example Title
  </a>
</h3>
`;

exports[`ItemTitle should render a heading tag with a set alignment. 1`] = `
<h3
  className="wordcamp-block__item-title"
  style={
    Object {
      "textAlign": "right",
    }
  }
>
  Example Title
</h3>
`;

exports[`ItemTitle should render a heading tag with the default level. 1`] = `
<h3
  className="wordcamp-block__item-title"
  style={Object {}}
>
  Example Title
</h3>
`;
